bpf/process/types/basic.h

```diff
@@ -2226,7 +2227,18 @@ generic_output(void *ctx, struct bpf_map_def *heap)
 		     :
 		     : [total] "+r"(total)
 		     :);
-	perf_event_output(ctx, &tcpmon_map, BPF_F_CURRENT_CPU, e, total);
+	ret = perf_event_output(ctx, &tcpmon_map, BPF_F_CURRENT_CPU, e, total);
+	switch (e->common.op) {
+	case MSG_OP_GENERIC_TRACEPOINT:
+		PERF_METRICS(METRIC_TRACEPOINT, ret);
+		break;
+	case MSG_OP_GENERIC_KPROBE:
+		PERF_METRICS(METRIC_KPROBE, ret);
+		break;
+	case MSG_OP_GENERIC_UPROBE:
+		PERF_METRICS(METRIC_UPROBE, ret);
+		break;
+	}
```

Comment:
maybe 

```
switch () {
     case TRACEPOINT:
       metric = METRIC_TRACEPOINT
     case KPROBE
       metric METRIC_KPROBE
}
// send event?
```

Reason is that I don't want to rely on compiler optimization to add a single call to the function.

Reply:
Fixed